spec/models/trip_spec.rb

```diff
@@ -0,0 +1,28 @@
+require "rails_helper"
+
+describe Trip do
+  describe "validations" do
+    it { is_expected.to validate_presence_of(:total_girls) }
+    it { is_expected.to validate_presence_of(:total_boys) }
+    it { is_expected.to validate_presence_of(:total_teachers) }
+    it { is_expected.to validate_presence_of(:total_bus_drivers) }
+    it { is_expected.to validate_presence_of(:start_date) }
+    it { is_expected.to validate_presence_of(:end_date) }
+  end
+
+  context "when start_date is in the past" do
+    it "is not valid" do
+      trip = build(:trip, start_date: 1.month.ago)
+
+      expect(trip).not_to be_valid
+    end
+  end
+
+  context "when start_date is after end_date" do
+    it "is not valid" do
+      trip = build(:trip, start_date: 1.month.ago, end_date: 2.month.ago)
```

Comment:
Maybe change this to be:
`build(:trip, start_date: 1.month.from_now, end_date: 2.month.ago)`
Just so we are sure we trigger the correct error message.

You should also check for the error messages returned

Reply:
Done both in fixup `85d5c3b'